### PR TITLE
Project Todo - Anna Lindgren

### PR DESCRIPTION
**Pull request by magicvilla**
Link to my deployed project: https://what-is-up-todo.netlify.app/

My reflections on how this week's project turned out:
I am so happy with my project this week. 
I had time to structure and plan which made me less stressed!

Things I'd like to have clarified or explained in more detail:

### CODE REVIEW

File: code/src/components/TodoContainer.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import '../css/TodoContainer.css'
+
+import TodoHeader from './TodoHeader'
+import TodoForm from './TodoForm'
+import TodoList from './TodoList'
+import TodoStartPage from './TodoStartPage'
+import TodoNavbar from './TodoNavbar'
+
```
Comment (M4L5) [inline code/src/components/TodoContainer.js:10]: Nicely structure of the imports :)

File: code/src/components/TodoContainer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                <p>Icons made by 
+                    <a href="https://www.freepik.com" title="Freepik"> Freepik</a> from
+                    <a href="https://www.flaticon.com/" title="Flaticon"> www.flaticon.com</a>
+                </p>
+            </footer>
+        </main>
+    )
+}
+
+export default TodoContainer
```
Comment (M4L5) [inline code/src/components/TodoContainer.js:33]: I like that you've done a todo container and added the components here. I also like that you used the ternary operator to either show startpage or todolist depending on if the user has made a todo. I might add that to my own later ;)

File: code/src/components/TodoForm.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        //dispatch an action to add new todo item
+        dispatch(
+            todos.actions.addTodo({
+                id: uniqid(),
+                description: value,
+                isComplete: false
+            })
+        )
+        //Clear input
+        setValue('')
```
Comment (M4L5) [inline code/src/components/TodoForm.js:30]: If you don't need the comments you can remove them. (Even though I find them helpful and has made the same project) 😃

File: code/src/components/TodoForm.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            />
+            <Button
+                variant="contained"
+                color="primary"
+                type="submit"
+                size="small"
+                startIcon={<SaveIcon />}
+                aria-label="add todo"
+            >
+                Add Task
```
Comment (M4L5) [inline code/src/components/TodoForm.js:53]: Super nice with the textfield and button from material ui! 😍

File: code/src/components/TodoHeader.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+const TodoHeader = () => {
+    const today = moment()
+
+    const todoItems = useSelector((store) => store.todos.items)
+    //Amount of todos in list
+    const numberOfTodos = todoItems.length
+    //Filter isComplete todos in array add length in return
+    const completedTodos = todoItems.filter(todo => todo.isComplete)
+
```
Comment (M4L5) [inline code/src/components/TodoHeader.js:18]: remove comments if you don't need them :)

File: code/src/components/TodoHeader.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    const dispatch = useDispatch()
+
+    const onClickComplete = () => {
+        dispatch(todos.actions.completeAllTodo())
```
Comment (M4L5) [inline code/src/components/TodoHeader.js:22]: Like this function!

File: code/src/components/TodoHeader.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    return (
+        <header className="header-wrapper">
+            <div className="header-title-container">
+            <h1>What's up today?</h1>
+            </div>
```
Comment (M4L5) [inline code/src/components/TodoHeader.js:29]: Just a small comment on indentation here. You could tab the h1 one step 👯

File: code/src/components/TodoList.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import React from 'react'
+import { useSelector, useDispatch } from 'react-redux'
+import Checkbox from '@material-ui/core/Checkbox';
+import IconButton from '@material-ui/core/IconButton'
+import DeleteIcon from '@material-ui/icons/Delete'
+import '../css/TodoList.css'
```
Comment (M4L5) [inline code/src/components/TodoList.js:6]: Not entirely sure, but I think you can seperate the imports for the material-ui from the two react imports to make it more readable :)

File: code/src/components/TodoNavbar.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+            <div className="menu-icon-container">
+                <img
+                    src={menu}
+                    alt="menu icon"
+                />
+            </div>
+            <div class="other-icon-container">
+                <img 
+                    src={loupe}
+                    alt="search icon"
```
Comment (M4L5) [inline code/src/components/TodoNavbar.js:21]: maybe another class name for the div that describes the icons more could be easier is someone else is reading your code :) Maybe something including notifications and search or tool-icons or something :)

File: code/src/reducers/todos.js [hunk 10]
```diff
@@ -0,0 +1,67 @@
+//Slice method
+import { createSlice } from '@reduxjs/toolkit'
+
+//Object with three properties: name, initialstate, reducers
+const todos = createSlice({
+    name: 'todos',
```
Comment (M4L5) [inline code/src/reducers/todos.js:6]: remove comments if not needed :)

File: code/src/reducers/todos.js [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+            const updatedItems = store.items.map(todo => {
+                if (todo.id === action.payload) {
+                    //change false to true
+                    //return current todo as it was (id and description) but with isComplete oposite to what previous state was. 
+                    return {
+                        ...todo,
+                        isComplete: !todo.isComplete
+                    }
+                } else {
+                    //leaves it as it was
```
Comment (M4L5) [inline code/src/reducers/todos.js:58]: remove comments if not needed :) I like them though cause it makes it easier for me as newbie to read ;)

**Review by M4L5 — COMMENTED** [review]
Anna, I REALLY like your todo list. I think the code is nicely structured and styling is really neat. If I had made my project as good as you I would have been very proud. You have reached the blue requirements and some of the red. Great job and nice to read the readme that you didn't feel as stressed and that you structured your work in a good way.

File: code/src/components/TodoContainer.js [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+            <footer className="footer-wrapper">
+                <p>Anna Lindgren - Technigo Bootcamp 2021</p>
+                <p>Icons made by 
+                    <a href="https://www.freepik.com" title="Freepik"> Freepik</a> from
+                    <a href="https://www.flaticon.com/" title="Flaticon"> www.flaticon.com</a>
+                </p>
+            </footer>
+        </main>
+    )
+}
```
Comment (KarolCislowski) [inline code/src/components/TodoContainer.js:31]: Please check your Prettier and configure it to have always 2 space deep indentation.

File: code/src/components/TodoContainer.js [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+                <TodoForm />
+                {totalCountTodo === 0 ? <TodoStartPage /> : <TodoList />}
+            </section>
+            <footer className="footer-wrapper">
+                <p>Anna Lindgren - Technigo Bootcamp 2021</p>
+                <p>Icons made by 
+                    <a href="https://www.freepik.com" title="Freepik"> Freepik</a> from
+                    <a href="https://www.flaticon.com/" title="Flaticon"> www.flaticon.com</a>
+                </p>
+            </footer>
```
Comment (KarolCislowski) [inline code/src/components/TodoContainer.js:28]: It is a great candidate to become separate component

File: code/src/components/TodoNavbar.js [hunk 14]
```diff
@@ -0,0 +1,32 @@
+import React from 'react'
+
+
```
Comment (KarolCislowski) [inline code/src/components/TodoNavbar.js:3]: 1 line gap is enough 😸

File: code/src/reducers/todos.js [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+                })
+                store.items = completedItems
+            }
+        },
+        
+        //Redux is always passing store as argument in order to update it
+        //add action to know which task to toggle.
+        toggleComplete: (store, action) => {
+            // console.log(store, action)
+            // console.log(action)
```
Comment (KarolCislowski) [inline code/src/reducers/todos.js:47]: commented out parts of code should be removed before requesting the pull.

**Review by KarolCislowski — APPROVED** [review]
This is a very nice app 🌈 
Please remember to clean your code from commented-out blocks of code 👮🏻

**Pull request closed without merging**